pyxnat/core/resources.py

```diff
@@ -1700,15 +1734,31 @@ def shares(self, id_filter='*'):
         return Projects(join_uri(self._uri, 'projects'),
                         self._intf, id_filter)
 
-    def share(self, project):
+    def share(self, project, label=None, primary=False):
         """ Share this assessor with another project.
 
             Parameters
             ----------
                 project: string
                     The other project name.
-        """
-        self._intf._exec(join_uri(self._uri, 'projects', project), 'PUT')
+                label: string
+                    New label of assessor. If empty the label of the experiment will be reused
+                primary: boolean
+                    If True, the primary ownership of the assessor will be changed from the original project to the
+                    new project. Logged-in user must have ownership permissions in both projects to change the primary
+                    ownership.
+        """
+        options = []
+        if label:
+            options.append('label=%s' % label)
+        else:
+            options.append('label=%s' % self._urn)
```

Comment:
Use the label instead of XNAT's unique identifier (or URN):
```suggestion
            options.append('label=%s' % self.label())
```